### DISCUSSION TITLE
Does the Crud Service support Unique Compound Indexes?

### DISCUSSION THREAD

**Discussion by alexBonadei** [mia-platform/community#566] [Q&A]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

1.108.2

### Description

Hi, I need to set a Unique Compound Index on two fields of my collection.  https://www.mongodb.com/docs/manual/core/index-unique/#unique-compound-index

I tried to do something like this:
![Screenshot from 2024-10-21 14-36-20](https://github.com/user-attachments/assets/dce1a597-067e-4e3e-82f5-f4a489369ee1)

but looks that it's not working.

I haven't found any reference about Unique Compound Index in the docs and in the changelog, so probably it's not supported.

If so, what is the pourpose for adding New Fields in the index?

Thank you in advance.

**Comment by malta895** [comment]
Hi @alexBonadei,
what you are asking should be actually supported.

How did you verify it's not working? Do you have direct access to the DB to verify that the index has been created?

Is the CRUD service up and running after the deployment, or does it report some error?
It is possible that the collection contains documents that violate the unique constraint, created before the introduction of this index. The CRUD service does not start in this case, reporting the related error in the logs.

**Reply by alexBonadei** [reply]
Hi @malta895,
I checked in the DB, and that index is missing.

We made some tests (version 7.0.29):

We tried to add other indexes, that were conflicting with the documents in the collection. The pod was up and running, but the index was not created.

We then dropped the collection and the indexes were correctly set.

Looks that the crud-service doesn't set the index when there are conflicts. Which is the opposite behavior that we were expecting (crud crashed if there are conflicts).
The fact is that this error is silent, so it's not easy to understand immediatly. Maybe it can be better documented, also the part about compound indexes that is very useful.

Thank you again!

**Reply by danibix95** [reply]
Hi, which version of CRUD service is currently deployed in your project?  
Starting from version `v7.2.0`, the service does not start in case it encounters an error creating an index, providing a log describing the issue.

**Reply by alexBonadei** [reply]
It's 7.0.2. (I made a typo before).
Ok then it makes sense. I will upgrade it asap.

Thank you!

**Reply by fredmaggiowski** [reply]
> Starting from version v7.2.0, the service does not start in case it encounters an error creating an index, providing a log describing the issue.

This was also the behaviour for versions prior v7 🤔.

Does this mean there was a regression in between the two versions? I didn't see it in the release note, I think it would be best to add it more explicitly
